FOP tests fail in Quarkus Platform

```
[INFO] 2020-10-19 13:00:50,531 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /fop/post?userConfigURL=file%3Afile%3A%2Fhome%2Fdata%2Fm2%2Frepository%2Forg%2Fapache%2Fcamel%2Fquarkus%2Fcamel-quarkus-integration-test-fop-rpkgtests%2F1.3.0%2Fcamel-quarkus-integration-test-fop-rpkgtests-1.3.0.jar%21%2Fmycfg.xml failed, error id: ee5724b0-3736-443d-9358-cad1225305eb-1: org.jboss.resteasy.spi.UnhandledException: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: fop://application/pdf?userConfigURL=file:file:/home/data/m2/repository/org/apache/camel/quarkus/camel-quarkus-integration-test-fop-rpkgtests/1.3.0/camel-quarkus-integration-test-fop-rpkgtests-1.3.0.jar!/mycfg.xml due to: java.io.FileNotFoundException: file:/home/data/m2/repository/org/apache/camel/quarkus/camel-quarkus-integration-test-fop-rpkgtests/1.3.0/camel-quarkus-integration-test-fop-rpkgtests-1.3.0.jar!/mycfg.xml (No such file or directory)
[INFO]  at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
[INFO]  at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
[INFO]  at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
[INFO]  at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
[INFO]  at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
[INFO]  at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
[INFO]  at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
[INFO]  at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
[INFO]  at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
[INFO]  at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
[INFO]  at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:131)
[INFO]  at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:37)
[INFO]  at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:94)
[INFO]  at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[INFO]  at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[INFO]  at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[INFO]  at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[INFO]  at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
[INFO]  at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
[INFO]  at java.base/java.lang.Thread.run(Thread.java:834)
[INFO]  at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[INFO] Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: fop://application/pdf?userConfigURL=file:file:/home/data/m2/repository/org/apache/camel/quarkus/camel-quarkus-integration-test-fop-rpkgtests/1.3.0/camel-quarkus-integration-test-fop-rpkgtests-1.3.0.jar!/mycfg.xml due to: java.io.FileNotFoundException: file:/home/data/m2/repository/org/apache/camel/quarkus/camel-quarkus-integration-test-fop-rpkgtests/1.3.0/camel-quarkus-integration-test-fop-rpkgtests-1.3.0.jar!/mycfg.xml (No such file or directory)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:896)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.getEndpoint(AbstractCamelContext.java:777)
[INFO]  at org.apache.camel.impl.engine.DefaultProducerTemplate.resolveMandatoryEndpoint(DefaultProducerTemplate.java:576)
[INFO]  at org.apache.camel.impl.engine.DefaultProducerTemplate.requestBody(DefaultProducerTemplate.java:410)
[INFO]  at org.apache.camel.quarkus.component.fop.it.FopResource.post(FopResource.java:55)
[INFO]  at org.apache.camel.quarkus.component.fop.it.FopResource_ClientProxy.post(FopResource_ClientProxy.zig:237)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[INFO]  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO]  at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[INFO]  at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
[INFO]  at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
[INFO]  at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:643)
[INFO]  at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:507)
[INFO]  at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:457)
[INFO]  at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
[INFO]  at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:459)
[INFO]  at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:419)
[INFO]  at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:393)
[INFO]  at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
[INFO]  at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
[INFO]  ... 17 more
[INFO] Caused by: org.apache.camel.RuntimeCamelException: java.io.FileNotFoundException: file:/home/data/m2/repository/org/apache/camel/quarkus/camel-quarkus-integration-test-fop-rpkgtests/1.3.0/camel-quarkus-integration-test-fop-rpkgtests-1.3.0.jar!/mycfg.xml (No such file or directory)
[INFO]  at org.apache.camel.RuntimeCamelException.wrapRuntimeException(RuntimeCamelException.java:66)
[INFO]  at org.apache.camel.support.service.BaseService.doFail(BaseService.java:409)
[INFO]  at org.apache.camel.support.service.BaseService.fail(BaseService.java:338)
[INFO]  at org.apache.camel.support.service.BaseService.start(BaseService.java:128)
[INFO]  at org.apache.camel.support.service.ServiceHelper.startService(ServiceHelper.java:84)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.deferStartService(AbstractCamelContext.java:1499)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.deferStartService(AbstractCamelContext.java:1476)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.internalAddService(AbstractCamelContext.java:1388)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.addService(AbstractCamelContext.java:1327)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.addService(AbstractCamelContext.java:1322)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.addService(AbstractCamelContext.java:1317)
[INFO]  at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:880)
[INFO]  ... 37 more
[INFO] Caused by: java.io.FileNotFoundException: file:/home/data/m2/repository/org/apache/camel/quarkus/camel-quarkus-integration-test-fop-rpkgtests/1.3.0/camel-quarkus-integration-test-fop-rpkgtests-1.3.0.jar!/mycfg.xml (No such file or directory)
[INFO]  at java.base/java.io.FileInputStream.open0(Native Method)
[INFO]  at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
[INFO]  at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
[INFO]  at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
[INFO]  at org.apache.camel.support.ResourceHelper.resolveResourceAsInputStream(ResourceHelper.java:166)
[INFO]  at org.apache.camel.support.ResourceHelper.resolveMandatoryResourceAsInputStream(ResourceHelper.java:141)
[INFO]  at org.apache.camel.component.fop.FopEndpoint.doStart(FopEndpoint.java:121)
[INFO]  at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
[INFO]  ... 45 more
```

--- a/integration-tests/fop/src/test/java/org/apache/camel/quarkus/component/fop/it/FopTest.java
+++ b/integration-tests/fop/src/test/java/org/apache/camel/quarkus/component/fop/it/FopTest.java
@@ -20,7 +20,11 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.io.StringWriter;
 import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.function.Function;
+import java.util.stream.Stream;
 
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
@@ -29,6 +33,7 @@ import io.restassured.response.ExtractableResponse;
 import io.restassured.specification.RequestSpecification;
 import org.apache.pdfbox.pdmodel.PDDocument;
 import org.apache.pdfbox.text.PDFTextStripper;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
@@ -37,6 +42,31 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 class FopTest {
 
     public static final String MSG = "hello";
+    private static Path tmpDir;
+
+    @BeforeAll
+    static void copyResources() throws IOException {
+        /*
+         * Local files are not available when this test is run in Quarkus Platform.
+         * ppalaga was not able to find a way to make FOP load the font declared in mycfg.xml from the classpath
+         * As a workaround, the resources are simply copied from the class loader to target/tmp directory
+         * before the test starts
+         */
+        tmpDir = Paths.get("target/tmp");
+        Files.createDirectories(tmpDir);
+        final ClassLoader cl = FopTest.class.getClassLoader();
+        Stream.of("Freedom-10eM.ttf", "mycfg.xml")
+                .forEach(resource -> {
+                    final Path target = tmpDir.resolve(resource);
+                    if (!Files.exists(target)) {
+                        try (InputStream in = cl.getResourceAsStream(resource)) {
+                            Files.copy(in, target);
+                        } catch (IOException e) {
+                            throw new RuntimeException("Could not read resource " + resource, e);
+                        }
+                    }
+                });
+    }
 
     @Test
     public void convertToPdf() throws IOException {
@@ -45,14 +75,15 @@ class FopTest {
 
     @Test
     public void convertToPdfWithCustomFont() throws IOException {
-        convertToPdf(msg -> decorateTextWithXSLFO(msg, "Freedom"), "/mycfg.xml");
+        convertToPdf(msg -> decorateTextWithXSLFO(msg, "Freedom"),
+                tmpDir.resolve("mycfg.xml").toAbsolutePath().toUri().toString());
     }
 
     private void convertToPdf(Function<String, String> msgCreator, String userConfigFile) throws IOException {
         RequestSpecification requestSpecification = RestAssured.given()
                 .contentType(ContentType.XML);
         if (userConfigFile != null) {
-            requestSpecification.queryParam("userConfigURL", "file:" + getClass().getResource(userConfigFile).getFile());
+            requestSpecification.queryParam("userConfigURL", userConfigFile);
         }
         ExtractableResponse response = requestSpecification
                 .body(msgCreator.apply(MSG))

